app/gui/src/project-view/components/GraphEditor/widgets/WidgetTypeCast.vue

```diff
@@ -0,0 +1,60 @@
+<script setup lang="ts">
+import NodeWidget from '@/components/GraphEditor/NodeWidget.vue'
+import { defineWidget, Score, WidgetInput, widgetProps } from '@/providers/widgetRegistry'
+import { injectWidgetTree } from '@/providers/widgetTree'
+import { Ast } from '@/util/ast'
+import { computed } from 'vue'
+import { IsTypeCastKey } from './WidgetTypeCastPort.vue'
+
+const props = defineProps(widgetProps(widgetDefinition))
+const tree = injectWidgetTree()
+
+const isSelfArgument = computed(() => {
+  if (!(props.input.value instanceof Ast.TypeAnnotated)) return false
+  const selfArgumentId = tree.primaryApplication.potentialSelfArgument
+  return props.input.value.expression.id === selfArgumentId
+})
+
+const expressionInput = computed(() => {
+  if (props.input.value instanceof Ast.TypeAnnotated)
+    return WidgetInput.FromAst(props.input.value.expression)
+  else return undefined
+})
+const typeNodeInput = computed(() => {
+  if (props.input.value instanceof Ast.TypeAnnotated)
+    return WidgetInput.FromAst(props.input.value.typeNode)
+  else return undefined
+})
+</script>
+
+<script lang="ts">
+export const widgetDefinition = defineWidget(
+  IsTypeCastKey,
```

Comment:
We don't check if the value is actually `true`

If we don't care, because it's always true, then the field's type should also be `true` (instead of `boolean`)

Reply:
Done